worlds/yoshisisland/setup_bosses.py

```diff
@@ -0,0 +1,16 @@
+from BaseClasses import CollectionState
+
+class BossReqs:
+    player: int
+
+
+    def __init__(self, player: int, world):
+        self.player = player
+        self.castle_unlock = world.options.castle_open_condition.value
+        self.boss_unlock = world.options.castle_clear_condition.value

```

Comment:
```suggestion
    def __init__(self, world):
        self.player = player
        self.castle_unlock = world.options.castle_open_condition.value
        self.boss_unlock = world.options.castle_clear_condition.value

```

You don't seem to be referencing player here at all

Reply:
I'm trying to follow this suggestion, but it keeps causing failures

Reply:
I don't really see why it would. It's literally just removing an unused reference. Maybe the error is somewhere else? I recommend seeing what errors are thrown, checking the traceback, and using that to determine how to fix it.